src/MemoryMgmt.chpl

```diff
@@ -74,11 +84,51 @@ module MemoryMgmt {
             if line.find('MemAvailable:') >= 0 {
                 var splits = line.split('MemAvailable:');
                 memAvail = splits[1].strip().strip(' kB'):uint(64);
+                break;
             }
         }
 
         return (AutoMath.round(availableMemoryPct/100 * memAvail)*1000):uint(64);
     }
+    
+    proc getTotalMemory() : uint(64) throws {
+        var aFile = open('/proc/meminfo', ioMode.r);
+        var lines = aFile.reader().lines();
+        var line : string;
+
+        var totalMem:uint(64);
+
+        for line in lines do {
+            if line.find('MemTotal:') >= 0 {
+                var splits = line.split('MemTotal:');
+                totalMem = splits[1].strip().strip(' kB'):uint(64);
+                break;
+            }
+        }
+
+        return totalMem*1000:uint(64);
+    }
+    
+    proc getLocaleMemoryStatuses() throws {
+        var memStatuses: [0..numLocales-1] LocaleMemoryStatus;
+        
+        coforall loc in Locales with (ref memStatuses) {
+            on loc {
+                var availMem = getAvailMemory();
+                var totalMem = getTotalMemory();
+                var pctAvailMem = (availMem:real/totalMem)*100:int;
+
+                memStatuses[here.id] = new LocaleMemoryStatus(total_mem=totalMem,
+                                                              avail_mem=availMem,
+                                                              pct_avail_mem=pctAvailMem:int,
+                                                              arkouda_mem_alloc=getArkoudaMemAlloc(),
+                                                              mem_used=memoryUsed(),
+                                                              locale_id=here.id,
+                                                              locale_hostname=here.hostname);                                      
+            }
+        }
+        return memStatuses;
```

Comment:
I was originally trying to add `memStatuses` to the looping var to make things more clear and avoid having to do the `ref` with intent (since that can be racy and cause problems). Something like the following
```chapel
var memStatuses: [makeDistDom(numLocales-1)] LocaleMemoryStatus;

coforall (loc, m) in zip(Locales, memStatuses) {
    on loc {
        var availMem = getAvailMemory();
        var totalMem = getTotalMemory();
        var pctAvailMem = (availMem:real/totalMem)*100:int;

        m = new LocaleMemoryStatus(total_mem=totalMem,
                                  avail_mem=availMem,
                                  pct_avail_mem=pctAvailMem:int,
                                  arkouda_mem_alloc=getArkoudaMemAlloc(),
                                  mem_used=memoryUsed(),
                                  locale_id=here.id,
                                  locale_hostname=here.hostname);                                      
    }
}
// localize memStatuses somehow so it lives on locale0 before returning
return memStatuses;
```
but that requires we distribute and localize `memStatuses`. I think since we only ever access `memStatuses[here.id]`, we won't have any race conditions because two loop iterations won't touch the same section at the same time.

So this is just a roundabout way of saying I think what's here should be fine lol